src/routes/bounce_rules_detailed.jsx

```diff
@@ -26,8 +36,126 @@ export default class BounceRuleDetailedPage extends React.Component {
     }
   }
 
+  onChangeRule(event, field) {
+    const { currentRule } = this.state;
+    const newRule = Object.assign({}, currentRule);
```

Comment:
After we refactor this, we can do like this.setState({ currentRule: { ...currentRule, [id]: value } }) rather than Object.assign({}, currentRule)

Reply:
ahh i was previously trying to do this but I didn't realize it was because I did not have babel set properly, thanks!